eclair-core/src/main/resources/reference.conf

```diff
@@ -149,6 +149,11 @@ eclair {
       // Delay enforcement of channel fee updates
       enforcement-delay = 10 minutes
     }
+
+    async-payments {
+        enabled = false
```

Comment:
fixed in [96acc3a](https://github.com/ACINQ/eclair/pull/2435/commits/96acc3aaff05cac3c81714c865b6745e5b70df3a).

I went with `relay.timeout`, but was wondering if perhaps it should be more explicitly called `relay.async-payment-timeout`. Any preference?

Reply:
I would be more specific and name it `relay.async-payment-timeout`.
Otherwise it would conflict with types of "timeout".

For example, when we cannot relay a payment (because the next peer is offline or we don't have liquidity), we currently instantly fail the incoming htlc. But we could instead wait a bit and retry a few times before failing the incoming htlc. That feature would need its own timeout, which would be named differently.

Reply:
Since changes to the configuration file are a pain (we need to detect the old setting and warn users), better to get the naming right from the start ;)

Reply:
Update: before renaming this, it's worth investigating this comment: https://github.com/ACINQ/eclair/pull/2435#discussion_r980290178

If we create a dedicated config section, the name `timeout` is good enough.

Reply:
Good point - it seems like there is room to expand on this. A new `async-payments` section (AsyncPaymentsParams case class) within `relay` (and RelayParams) is justified.

Reply:
Replaced clock based timeout in the `relay` section with a new `async-payments` section and two block delta timeouts:
```
    async-payments {
      // Maximum number of blocks to hold an async payment while waiting to receive a trigger from the receiver
      hold-timeout-blocks = 1008
      // Number of blocks before the incoming HTLC expires that an async payment must be triggered by the receiver
      cancel-safety-before-timeout-blocks = 144
    }
```
 fixed in [fcdc394](https://github.com/ACINQ/eclair/pull/2435/commits/fcdc394cb2751d7408318afb437cf49d18389d08)